core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/BinPackingNodeAllocatorService.java

```diff
@@ -303,6 +317,46 @@ public void close()
         //      and that can be done before all leases are yet returned from running (soon to be failed) tasks.
     }
 
+    @Managed
+    @Nested
+    public StatsHolder getStats()
```

Comment:
Worth a code comment why this has to be same instance, why stats object inside can change.

(Does it work with /metrics ?)